### PR TITLE
Features/new auth

### PR DESCRIPTION
**Pull request by gsfk**
First pass at adding authz to beacon: response to main `/individuals` endpoint varies with user permissions:
- users with valid token and PERMISSION_QUERY_DATA get full response to queries
- additionally, if they have PERMISSION_DOWNLOAD_DATA they can get valid download links for DRS. 

Users without permissions will get the default response only, whether they have a token or not (response currently hardcoded to counts, configurable response coming soon) 

The `/individuals/<id>` endpoint requires a token with valid permissions, all other users receive 401 / 403.

### CODE REVIEW

File: bento_beacon/utils/handover_utils.py [hunk 1]
```diff
@@ -1,68 +1,36 @@
...
-    external_url_components = drs_external_url_components()
-    path = external_url_components.path + "/objects/" + id + "/download"
+
+def drs_url_components():
+    return urlsplit(current_app.config["DRS_URL"])
+
+
+def drs_file_link_for_id(id):
+    url_components = drs_url_components()
+    path = url_components.path + "/objects/" + id + "/download"
```
Comment (davidlougheed) [inline bento_beacon/utils/handover_utils.py:18]: this should already be built in the DRS response I think?

Comment (davidlougheed) [reply bento_beacon/utils/handover_utils.py:18]: (via the `access_methods` array)

File: bento_beacon/authz/middleware.py [hunk 2]
```diff
@@ -27,3 +32,18 @@
...
 PERMISSION_QUERY_DATA = "query:data"
 PERMISSION_DOWNLOAD_DATA = "download:data"
+
+
+def check_permissions(permissions: list[str]) -> bool:
+    auth_res = authz_middleware.authz_post(request, "/policy/evaluate", body={
+        "requested_resource": {"everything": True},
+        "required_permissions": permissions,
+        },
+        headers_getter=auth_header_getter,  # currently same as middleware default, so can be removed
```
Comment (davidlougheed) [inline bento_beacon/authz/middleware.py:42]: I think this is safe to remove

**Review by davidlougheed — CHANGES_REQUESTED** [review]

File: bento_beacon/utils/handover_utils.py (hunk 1 above)
Comment (gsfk) [reply bento_beacon/utils/handover_utils.py:18]: ah, yes, thanks for this, will refactor

File: bento_beacon/authz/middleware.py [hunk 3]
```diff
@@ -1,6 +1,8 @@
+from flask import request
 from bento_lib.auth.middleware.flask import FlaskAuthMiddleware
-from .config_files.config import Config
-from .utils.beacon_response import build_response_meta
+from ..config_files.config import Config
+from ..utils.beacon_response import build_response_meta
+from .headers import auth_header_getter
```
Comment (davidlougheed) [inline bento_beacon/authz/middleware.py:5]: this import now goes unused

File: bento_beacon/authz/headers.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from flask import request, Request
```
Comment (davidlougheed) [inline bento_beacon/authz/headers.py:1]: this file is just unused stuff, but we could leave it if you intend to do some non-bearer-token stuff in the future

Comment (gsfk) [reply bento_beacon/authz/headers.py:1]: yeah, it's going to be in the katsu / gohan stuff

**Review by davidlougheed — APPROVED** [review]
lgtm

**Pull request merged**